test/cases_filter_matching_callbacks/4.callbacks.c

```
@@ -0,0 +1,45 @@
+#include <metababel/metababel.h>
+#include <stdbool.h>
+#include <assert.h>
+
+struct data_s {
+  uint64_t condition_calls_count;
+  uint64_t callback_calls_count;
+};
+
+typedef struct data_s data_t;
+
+void btx_initialize_usr_data(void *btx_handle, void **usr_data) {
+  *usr_data = calloc(1, sizeof(data_t));
+}
+
+void btx_finalize_usr_data(void *btx_handle, void *usr_data) {
+  data_t *data = (data_t *)usr_data;
+
+  assert(data->condition_calls_count == 4);
+  assert(data->callback_calls_count == 4);
+
+  free(data);
+}
+
+static void btx_condition(void *btx_handle, void *usr_data, bool *matched, const char* event_class_name)
+{
+  data_t *data = (data_t *)usr_data;
+  *matched = (0 == strcmp(event_class_name, "event_1"));
+  if (*matched) data->condition_calls_count += 1;
+}
+
+static void btx_callback(void *btx_handle, void *usr_data, const char* event_class_name)
+{
+  data_t *data = (data_t *)usr_data;
+  if (0 == strcmp(event_class_name, "event_1")) data->callback_calls_count += 1;
```

I don't think you will need the `if` here.

---

But maybe raise an error if `event_class_name` is equal to `event_2`.